tests/cupy_tests/manipulation_tests/test_kind.py

```diff
@@ -10,6 +10,19 @@
 @testing.gpu
 class TestKind(unittest.TestCase):
 
+    @testing.for_all_dtypes()
+    def test_asarray_chkfinite(self, dtype):
+        a = [0, 4, 0, 5]
+        a_gpu = cupy.asarray_chkfinite(a, dtype=dtype)
+        a_cpu = numpy.asarray_chkfinite(a, dtype=dtype)
+        assert a_gpu.dtype == a_cpu.dtype
+
+    @unittest.expectedFailure
+    @testing.for_all_dtypes()
+    def test_asarray_chkfinite_non_finite_vals(self, dtype):
+        a = [-numpy.inf, 0., numpy.inf]
```

Comment:
Nitpick: Ideally we also want to include NaN, though it's probably shadowed by the first non-finite entry... 
```suggestion
        a = [-numpy.inf, 0., numpy.inf, numpy.nan]
```